pkg/envoy/registry/registry.go

```diff
@@ -10,38 +12,32 @@ func NewProxyRegistry(mapper ProxyServiceMapper, msgBroker *messaging.Broker) *P
 	return &ProxyRegistry{
 		ProxyServiceMapper: mapper,
 		msgBroker:          msgBroker,
-		connectedProxies:   make(map[string]*envoy.Proxy),
+		connectedProxies:   make(map[int64]*envoy.Proxy),
+		pubsub:             pubsub.New(0),
+		// ch 							: make(chan *envoy.Proxy, 10),
 	}
 }
 
 // RegisterProxy registers a newly connected proxy.
 func (pr *ProxyRegistry) RegisterProxy(proxy *envoy.Proxy) {
-	// TODO(#4950) check register request sequence before proceeding
-	uuid := proxy.UUID.String()
-	if pr.GetConnectedProxy(uuid) != nil {
-		log.Debug().Str("proxy", proxy.String()).Msgf("Proxy %s already registered", proxy.String())
-		return
-	}
-
 	pr.mu.Lock()
 	defer pr.mu.Unlock()
-	pr.connectedProxies[uuid] = proxy
+	pr.connectedProxies[proxy.GetConnectionID()] = proxy
 	log.Debug().Str("proxy", proxy.String()).Msg("Registered new proxy")
```

Comment:
nit: this might be slightly misleading comment in since the proxy potentially already registered?

Reply:
fixed!